nmstatectl/nmstatectl.py

```diff
@@ -90,12 +116,49 @@ def _filter_interfaces(state, patterns):
     return showinterfaces
 
 
-def show(args):
-    state = netinfo.show()
-    if args.only != '*':
-        patterns = [p for p in args.only.split(',')]
+def _filter_state(state, whitelist):
+    if whitelist != '*':
+        patterns = [p for p in whitelist.split(',')]
         state['interfaces'] = _filter_interfaces(state, patterns)
+    return state
+
+
+def edit(args):
+    state = _filter_state(netinfo.show(), args.only)
+
+    editor = os.environ.get('EDITOR', 'vi')
+    if args.yaml:
+        suffix = '.yaml'
+        txtstate = pretty_yaml(state)
+    else:
+        suffix = '.json'
+        txtstate = pretty_json(state)
+    with tempfile.NamedTemporaryFile(suffix=suffix,
+                                     prefix='nmstate-') as statefile:
+        statefile.write(txtstate)
+        statefile.flush()
+
+        try:
+            subprocess.check_call([editor, statefile.name])
 
+            statefile.seek(0)
+
+            if args.yaml:
+                new_state = yaml.load(statefile)
+            else:
+                new_state = json.load(statefile)
```

Comment:
Please handle possible exceptions of `json.load()` and `yaml.load()`.

Reply:
Handle what exactly? Exploding when something is wrong sounds a good solution to me.

Reply:
For example, invalid json/yaml syntax.

Reply:
Ok, so how would you recommend handling it?

Reply:
```diff
diff --git a/nmstatectl/nmstatectl.py b/nmstatectl/nmstatectl.py
index 7241635..fffbd37 100644
--- a/nmstatectl/nmstatectl.py
+++ b/nmstatectl/nmstatectl.py
@@ -143,10 +143,23 @@ def edit(args):
             statefile.seek(0)
-            if args.yaml:
-                new_state = yaml.load(statefile)
-            else:
-                new_state = json.load(statefile)
+            try:
+                if args.yaml:
+                    new_state = yaml.load(statefile)
+                else:
+                    new_state = json.load(statefile)
+            except ValueError as e:
+                sys.stderr.write("ERROR: Invalid JSON syntax: %s\n" % e)
+                return 1
+            except yaml.parser.ParserError:
+                sys.stderr.write("ERROR: Invalid YAML syntax: %s\n" % e)
+                return 1
+
+            if 'interfaces' not in new_state:
+                sys.stderr.write("ERROR: Invalid JSON/YAML syntax: "
+                                 "should contain 'interfaces' entry.\n")
+                return 1
+
             print('Applying the following state: ')
             print_state(new_state, use_yaml=args.yaml)
```

Reply:
Should there be an extra case to catch empty files? You mentioned it in another comment IIRC.

Reply:
Oh. I was waiting a decision before coding anything out.

Reply:
I added your code as a separate commit.